packages/sanity/src/structure/comments/plugin/inspector/CommentsInspector.tsx

```diff
@@ -37,9 +39,9 @@ const RootLayer = styled(Layer)`
 `
 
 export function CommentsInspector(props: DocumentInspectorProps) {
-  const isEnabled = useCommentsEnabled()
+  const {enabled} = useCommentsEnabled()
 
-  if (!isEnabled) return null
+  if (!enabled) return null
```

Comment:
Do we need to check if `reason` is 'upsell' here as well? Otherwise, I think we will return `null` even though we want to show the inspector with the upsell UI since we only check the `enabled` value.

Reply:
enabled will be true when reason is `upsell` .